sqlalchemy_redshift/dialect.py

```diff
@@ -547,12 +543,11 @@ def create_connect_args(self, *args, **kwargs):
         return cargs, default_args
 
     def _get_table_or_view_names(self, relkind, connection, schema=None, **kw):
-        default_schema = inspect(connection).default_schema_name
         if not schema:
-            schema = default_schema
+            schema = self.default_schema_name
```

Comment:
Using this attribute definitely cleans up the code compared to passing a connection around and calling inspect, which is very nice.